File: itinerarymanagement.client/src/app/models/route/route-request.ts
export interface RouteRequest {
  startLat: number;
  startLon: number;
  endLat: number;
  endLon: number;
  vehicleType?: VehicleType;
}
export enum VehicleType {
  Pedestrian = 0,
  Car = 1,
  Bicycle = 2
}

File: ItineraryManagement.Server/Models/Route/RouteRequest.cs
using ItineraryManagement.Server.Helpers;

namespace ItineraryManagement.Server.Models.Route
{
    public class RouteRequest
    {
        public float StartLat { get; set; }
        public float StartLon { get; set; }
        public float EndLat { get; set; }
        public float EndLon { get; set; }
        public VehicleType VehicleType { get; set; } = VehicleType.Pedestrian;
    }
}
